File: ProyectoCsharp/Modelo/JefeDeAlmacen.cs
using System;

namespace Modelo
{
    public class JefeDeAlmacen : Usuario
    {
        private AlmacenPrincipal almacen;



        public JefeDeAlmacen(long id_usuario, string nombre, string apellidos, string correo, string dni, string username, string contraseña, string descripcion_permisos,
                             AlmacenPrincipal almac) : base(id_usuario, nombre, apellidos, correo, dni, username, contraseña, descripcion_permisos)
        {
            Almacen = almac;
        }

        public AlmacenPrincipal Almacen { get => almacen; set => almacen = value; }
    }

}

File: TrabajoJavaLp2/src/modelo/JefeDeAlmacen.java
package modelo;
public class JefeDeAlmacen extends Usuario{
	private Almacen almacen; 

	public JefeDeAlmacen(){
		almacen = new Almacen();
	}

	public JefeDeAlmacen(long id_usuario, String nombre, String apellidos, String correo, String codigo,
                    String username, String contrasena, String descripcion_permisos)
        {
            super(id_usuario, nombre,
                apellidos, correo, codigo, username, contrasena, descripcion_permisos);
            setAlmacen(almacen);
	}

	public Almacen getAlmacen(){
		return almacen;
	}

	public void setAlmacen(Almacen almacen){
		this.almacen = almacen;
	}
}